src/components/BInput/BInput.vue

```diff
@@ -0,0 +1,196 @@
+<template>
+  <div class="flex">
+    <label :for="id" :class="{ disabled: disabled }">
+      {{ label }}<span v-if="required" aria-label="Required">*</span>
+    </label>
+    <span v-if="type === 'currency'" :class="{ currencyinput: true, focus: hasFocus, disabled: disabled, invalid: errors.any() }" @click="$refs.input.focus()">
+      <strong>$</strong>
+      <input v-model="publicValue" v-bind="inputAttributes" v-validate="validations" @focus="hasFocus = true" @blur="hasFocus = false" ref="input" />
+    </span>
+    <input v-else v-model="publicValue" v-bind="inputAttributes" v-validate="validations" :class="{ invalid: errors.any() }" />
+    <template v-if="errors.any()">
```

Comment:
I think you can do this on the `span`. No need for `template tag`

Reply:
Agree about not needing template. What's your argument for reversing the span/div nesting?

Reply:
To be honest, it should probably just be a `span` or a `div` since it's not grouping content but just displaying text. Also, `span` is inline where `div` is block.

Reply:
agreed